### PR TITLE
better loss prints: add train_loss, val_loss, and val_acc during fitting

### PR DESCRIPTION
**Pull request by apiltamang**
Just better print statistics during learn.fit(..). Now, there's just more information, which otherwise could be confusing.

To Test:
- Just learn notebooks as is. Notice that at the end of each epoch, you get an output that looks as the follows:

screenshot 2017-11-04 16 46 39

### CODE REVIEW

**Comment by yanneta** [comment]
Hi,
There are a couple of problems here. We often don't use accuracy as the metric so "val_acc" would be misleading. If you read the code carefully, "metrics" is a list of arbitrary length. "metrics" can be defined by the user.  I am not sure the best way to solve it. 

Thank you,
Yannet

**Comment by apiltamang** [comment]
That is true. I think though that "train_loss", and "val_loss" would be fairly certain to be always applicable, no?

Anyways, I think there could be a design improvement here. Just printing out an array of numbers definitely seems a little crude. Let me know what a best possible alternate may be.

**Comment by yanneta** [comment]
At the moment I don't have a good suggestion. Let's keep thinking about it. I think that after one day of training you get used to the numbers.

**Comment by jph00** [comment]
I think this is fine as is, except that you should just write 'val_metrics:' at the end, and then show all metrics in the array there. You should test it to ensure it works correctly with multiple metrics.

**Comment by jph00** [comment]
@apiltamang are you planning to fix up this PR? Or should I just close it for now?

**Comment by apiltamang** [comment]
@jph00 
I can do a basic rewrite.. no problem.
As for the metrics, unfortunately, I'm not aware of all the metrics that can come by. So, definitely blindsided a bit. If it can be merged for now, I'll continue to test as I learn more of the possible metrics.

Anyways let me know.. it's not a big deal. If you'd rather close it, I'm fine.

**Comment by alexandrecc** [comment]
@apiltamang , imho, inspiration from keras could be useful :
https://github.com/fchollet/keras/blob/master/keras/metrics.py
https://github.com/fchollet/keras/blob/master/keras/losses.py

**Comment by oostopitre** [comment]
Can we display it as a header instead on every line? that way it does not waste space on the row and it pulls the headers from the metrics array?

**Comment by jph00** [comment]
@apiltamang you don't need to know what metrics might come (and indeed you can't, since the user can write their own!) That's why I suggested "just write 'val_metrics:' at the end, and then show all metrics in the array there" - that is, simply join the metrics up into a single string, just like I do now, and don't try to name each one individually. Does that make sense?

**Comment by jph00** [comment]
Closing this for now since it's not getting any action. Feel free to create a new PR fixing issues mentioned.
